test/parallel/test-stream2-httpclient-response-end.js

```diff
@@ -11,7 +11,9 @@ const server = http.createServer(function(req, res) {
     let data = '';
     res.on('readable', common.mustCall(function() {
       console.log('readable event');
-      data += res.read();
+      let input = res.read();
+      if (input)
```

Comment:
Ack. Learned something new.
So why is this flaky, the description sounds deterministic, `null` should always be concat to `data`